cpp/tests/groupby/group_count_scan_test.cpp

```diff
@@ -0,0 +1,213 @@
+}
+
+struct groupby_count_scan_string_test : public cudf::test::BaseFixture {
+};
+
+TEST_F(groupby_count_scan_string_test, basic)
+{
+  using K = int32_t;
+  using V = cudf::string_view;
+  using R = cudf::detail::target_type_t<V, aggregation::COUNT_ALL>;
+
+  // clang-format off
+  fixed_width_column_wrapper<K> keys        {   1,   3,   3,   5,   5,   0};
+  strings_column_wrapper        vals        { "1", "1", "1", "1", "1", "1"};
+
+  fixed_width_column_wrapper<K> expect_keys     {0, 1, 3, 3, 5, 5};
+  fixed_width_column_wrapper<R, int> expect_vals{0, 0, 0, 1, 0, 1};
+  // clang-format on
+
+  auto agg2 = cudf::make_count_aggregation(null_policy::INCLUDE);
+  test_single_scan(keys, vals, expect_keys, expect_vals, std::move(agg2));
+}
+
+template <typename T>
+struct FixedPointTestBothReps : public cudf::test::BaseFixture {
+};
+
+TYPED_TEST_CASE(FixedPointTestBothReps, cudf::test::FixedPointTypes);
+
+TYPED_TEST(FixedPointTestBothReps, GroupByCountScan)
+{
+  using namespace numeric;
+  using decimalXX  = TypeParam;
+  using RepType    = cudf::device_storage_type_t<decimalXX>;
+  using fp_wrapper = cudf::test::fixed_point_column_wrapper<RepType>;
+
+  using K = int32_t;
+  using V = decimalXX;
+  using R = cudf::detail::target_type_t<V, aggregation::COUNT_ALL>;
+
+  auto const scale = scale_type{-1};
+  // clang-format off
+  auto const keys  = fixed_width_column_wrapper<K>{1, 2, 3, 1, 2, 2, 1, 3, 3, 2};
+  auto const vals  = fp_wrapper{                  {0, 1, 2, 3, 4, 5, 6, 7, 8, 9}, scale};
+
+  auto const expect_keys = fixed_width_column_wrapper<K>{1, 1, 1, 2, 2, 2, 2, 3, 3, 3};
+  auto const expect_vals = fixed_width_column_wrapper<R, int>{0, 1, 2, 0, 1, 2, 3, 0, 1, 2};
+  // clang-format on
+
+  CUDF_EXPECT_THROW_MESSAGE(
+    test_single_scan(keys, vals, expect_keys, expect_vals, cudf::make_count_aggregation()),
+    "Unsupported groupby scan aggregation");
+
+  auto agg2 = cudf::make_count_aggregation(null_policy::INCLUDE);
+  test_single_scan(keys, vals, expect_keys, expect_vals, std::move(agg2));
+}
+
+struct groupby_dictionary_count_scan_test : public cudf::test::BaseFixture {
+};
+
+TEST_F(groupby_dictionary_count_scan_test, basic)
+{
+  using K = int32_t;
+  using V = std::string;
+  using R = cudf::detail::target_type_t<V, aggregation::COUNT_ALL>;
+
+  // clang-format off
+  strings_column_wrapper       keys{"1", "3", "3", "5", "5", "0"};
+  dictionary_column_wrapper<K> vals{ 1,   1,   1,   1,   1,   1};
+  strings_column_wrapper             expect_keys{"0", "1", "3", "3", "5", "5"};
+  fixed_width_column_wrapper<R, int> expect_vals{ 0,   0,   0,   1,   0,   1};
+  // clang-format on
+
+  auto agg1 = cudf::make_count_aggregation();
+  CUDF_EXPECT_THROW_MESSAGE(test_single_scan(keys, vals, expect_keys, expect_vals, std::move(agg1)),
+                            "Unsupported groupby scan aggregation");
+  test_single_scan(
+    keys, vals, expect_keys, expect_vals, cudf::make_count_aggregation(null_policy::INCLUDE));
+}
```

Comment:
Looks like dictionary is supported for `max_scan` and `min_scan`. Can you add some gtests for dictionary?

Reply:
No. dictionary is not supported yet for `max_scan` and `min_scan`.

Reply:
Ok, but the code in `group_max_scan.cu` in this PR has dictionary stuff in it
https://github.com/rapidsai/cudf/pull/7387/files#diff-79102e0a5e128a364ed61de1b4570dd8f1e35cedf903fafb65d82a9c171d6949R31-R33
As well as `group_min_scan.cu`
https://github.com/rapidsai/cudf/pull/7387/files#diff-669587e7204b4fe43a9adec0d891843a6a44317e0ab8a1d2b906259c3246aa1dR31-R33
Should we throw an exception here instead then?

Reply:
updated the code. It will throw exception now.

Reply:
I thought `count_scan` was working for dictionary.
I don't understand why this works
```
test_single_scan(keys, vals, expect_keys, expect_vals, cudf::make_count_aggregation(null_policy::INCLUDE));
```
and this throws an exception
```
test_single_scan(keys, vals, expect_keys, expect_vals, cudf::make_count_aggregation());
```